### ISSUE TITLE
Rails cache error

### ISSUE THREAD

**Issue by anderson** [meilisearch/meilisearch-rails#246]
**Description**
When Rails.cache attempting to use Marshal.dump on the result of a Model.search('*') call, a TypeError is encountered, specifically "singleton can't be dumped."

**Expected behavior**
The expected behavior is for Marshal.dump to successfully serialize the object returned by Model.search('*') without raising any errors.

**Current behavior**
```
irb(main):001:0> Marshal.dump(Model.search('*'))
(irb):1:in `dump': singleton can't be dumped (TypeError)
```

**Environment (please complete the following information):**
- OS: MacOS 13.2.1
- Meilisearch server version: 1.0.2
- meilisearch-rails version: v0.8.1 and main
- Rails version: 7.0.4.3

**Comment by brunoocasali** [comment]
Hi @anderson, thanks for your report. 

Before digging into your request, I would like to know why you want to cache the Meilisearch search response since our big value is to return relevant results quickly (usually under 50ms)...
If you want to prevent the database queries usually made after the Meili response, we could think of a way to do it.

**Comment by anderson** [comment]
Hi @brunoocasali,

Thank you for your response.

Meilisearch is super fast and does an amazing job. However, sometimes we need to handle a variety of scenarios involving database query results, which is why having compatibility with Rails caching can be a real lifesaver for performance.

I've submitted a PR  #247 with a simple solution for this issue. This should help us avoid extra database queries after the Meilisearch response and keep everything running smoothly.

Let me know if you have any further questions or concerns!

**Comment by brunoocasali** [comment]
Thanks a lot for the fix @anderson It will be released in v0.9.0
